src/Workspaces/CSharp/Portable/Simplification/Reducers/AbstractCSharpReducer.cs

```diff
@@ -14,5 +14,11 @@ internal abstract partial class AbstractCSharpReducer : AbstractReducer
         protected AbstractCSharpReducer(ObjectPool<IReductionRewriter> pool) : base(pool)
         {
         }
+
+        public sealed override bool IsApplicable(SimplifierOptions options)
+            => IsApplicable((CSharpSimplifierOptions)options);
+
+        protected virtual bool IsApplicable(CSharpSimplifierOptions options)
+            => true;
```

Comment:
don't love this.  would prefer abstract with every reducer having to thinka bout it.

Reply:
(note: if this was already our pattern, fine to leave as is.)

Reply:
Made abstract.